tripy/tripy/frontend/cache.py

```diff
@@ -0,0 +1,153 @@
+            trace (Trace): The trace to update.
+            tensor_map (Dict[int, str]): Mapping of tensor ids to names.
+            next_id (List[int]): Mutable list for tracking next tensor ID.
+            backup_map (Dict[int, str], optional): Mapping of original tensor names. Defaults to None.
+        """
+        # Update input names
+        for inp in trace.inputs:
+            inp.name = self._assign_tensor_name(inp, tensor_map, next_id, backup_map)
+
+        # Update operation input and output names
+        for op in trace.ops:
+            for inp in op.inputs:
+                inp.name = self._assign_tensor_name(inp, tensor_map, next_id, backup_map)
+            for out in op.outputs:
+                out.name = self._assign_tensor_name(out, tensor_map, next_id, backup_map)
+
+        # Update output names
+        for out in trace.outputs:
+            out.name = self._assign_tensor_name(out, tensor_map, next_id, backup_map)
+
+    def _normalize_trace(self, trace: "Trace") -> str:
+        """
+        Normalize the trace by renaming all tensor names while preserving the structure.
+
+        Args:
+            trace (Trace): The trace to normalize.
+
+        Returns:
+            str: Normalized trace as a string.
+        """
+        # Initialize maps and next tensor ID
+        tensor_map: Dict[int, str] = {}
+        backup_tensor_map: Dict[int, str] = {}
+        next_tensor_id: List[int] = [0]
+
+        # Clean trace names with sequential names
+        self._update_trace_names(trace, tensor_map, next_tensor_id, backup_tensor_map)
+
+        # Get normalized trace string
+        trace_str = str(trace)
+
+        # Restore original names
+        self._update_trace_names(trace, backup_tensor_map, next_tensor_id)
+
+        return trace_str
+
+    def _generate_key(self, trace: "Trace", devices: List["tripy.common.device"]) -> str:
+        """
+        Generate a unique key for a given trace and device configuration.
+
+        Args:
+            trace (Trace): The trace for which to generate the key.
+            devices (List[Device]): List of devices associated with the trace.
+
+        Returns:
+            str: A unique hash key representing the trace and devices.
+        """
+        normalized_trace = self._normalize_trace(trace)
+        key = normalized_trace + "\ndevices:\n" + "\n".join([str(device) for device in devices])
+        return hashlib.sha256(key.encode("utf-8")).hexdigest()
+
+    def get(self, trace: "Trace", devices: List["tripy.common.device"]) -> runtime.Executable:
+        """
+        Retrieve a cached executable for the given trace and devices.
+
+        Args:
+            trace (Trace): The trace used as a key.
+            devices (List[Device]): List of devices associated with the trace.
+
+        Returns:
+            Executable: The cached executable, or None if not found.
+        """
+        key = self._generate_key(trace, devices)
+        return self._cache.get(key, None)
```

Comment:
The default value for `get` is already None:
```suggestion
        return self._cache.get(key)
```